### PR TITLE
Fixed annotations and minor code problems.

### PR DESCRIPTION
**Pull request by badyto**
This will fix errors that are showing in IDE with code quality inspections enabled. 
Also fixes auto-completion of functions in IDE.

### CODE REVIEW

**Comment by peppeocchi** [comment]
@badyto thank you for this PR, can I just ask you to remove [this line](https://github.com/peppeocchi/php-cron-scheduler/pull/22/files#diff-e4f734dca0d6ac9903f887a7aa557b5dR5) and add it [here](https://github.com/peppeocchi/php-cron-scheduler/blob/master/src/GO/Job.php#L50)? I guess you've added it because of IDE completion.

I'll add comments if there is anything else I'd like to change, thank you.

File: src/GO/Job.php [hunk 1]
```diff
@@ -401,7 +402,8 @@ private function removeLockFile()
     /**
      * Execute a callable job.
      *
-     * @param  callable  $fn
+     * @param  callable $fn
```
Comment (peppeocchi) [inline src/GO/Job.php]: Can you please re-add the extra space? I'm using two spaces between the type and the var name everywhere else.

File: src/GO/Job.php [hunk 2]
```diff
@@ -498,7 +500,7 @@ private function finalise()
     /**
      * Email the output of the job, if any.
      *
-     * @return void
+     * @return bool
```
Comment (peppeocchi) [inline src/GO/Job.php]: I'd like to keep it as void, I guess there is no difference between returning nothing or returning always `true`, unless you have an argue that's better returning true.
PS: remove also the `return true;` at the end of this fn thanks

File: src/GO/Traits/Interval.php [hunk 3]
```diff
@@ -256,10 +256,10 @@ public function may($day = 1, $hour = 0, $minute = 0)
     /**
      * Set the execution time to every June.
      *
-     * @param  int\string  $day
-     * @param  int\string  $hour
-     * @param  int\string  $minute
-     * @return this
+     * @param int|string  $day
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between `@param` and the type?

File: src/GO/Traits/Interval.php [hunk 4]
```diff
@@ -269,10 +269,10 @@ public function june($day = 1, $hour = 0, $minute = 0)
     /**
      * Set the execution time to every July.
      *
-     * @param  int\string  $day
-     * @param  int\string  $hour
-     * @param  int\string  $minute
-     * @return this
+     * @param int|string  $day
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Traits/Interval.php [hunk 5]
```diff
@@ -282,10 +282,10 @@ public function july($day = 1, $hour = 0, $minute = 0)
     /**
      * Set the execution time to every August.
      *
-     * @param  int\string  $day
-     * @param  int\string  $hour
-     * @param  int\string  $minute
-     * @return this
+     * @param int|string  $day
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Traits/Interval.php [hunk 6]
```diff
@@ -295,10 +295,10 @@ public function august($day = 1, $hour = 0, $minute = 0)
     /**
      * Set the execution time to every September.
      *
-     * @param  int\string  $day
-     * @param  int\string  $hour
-     * @param  int\string  $minute
-     * @return this
+     * @param int|string  $day
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Traits/Interval.php [hunk 7]
```diff
@@ -308,10 +308,10 @@ public function september($day = 1, $hour = 0, $minute = 0)
     /**
      * Set the execution time to every October.
      *
-     * @param  int\string  $day
-     * @param  int\string  $hour
-     * @param  int\string  $minute
-     * @return this
+     * @param int|string  $day
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Traits/Interval.php [hunk 8]
```diff
@@ -321,10 +321,10 @@ public function october($day = 1, $hour = 0, $minute = 0)
     /**
      * Set the execution time to every November.
      *
-     * @param  int\string  $day
-     * @param  int\string  $hour
-     * @param  int\string  $minute
-     * @return this
+     * @param int|string  $day
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Traits/Interval.php [hunk 9]
```diff
@@ -334,10 +334,10 @@ public function november($day = 1, $hour = 0, $minute = 0)
     /**
      * Set the execution time to every December.
      *
-     * @param  int\string  $day
-     * @param  int\string  $hour
-     * @param  int\string  $minute
-     * @return this
+     * @param int|string  $day
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Traits/Interval.php [hunk 10]
```diff
@@ -347,11 +347,11 @@ public function december($day = 1, $hour = 0, $minute = 0)
     /**
      * Validate sequence of cron expression.
      *
-     * @param  int\string  $minute
-     * @param  int\string  $hour
-     * @param  int\string  $day
-     * @param  int\string  $month
-     * @param  int\string  $weekday
+     * @param int|string  $minute
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Traits/Interval.php [hunk 11]
```diff
@@ -368,7 +368,7 @@ private function validateCronSequence($minute = null, $hour = null, $day = null,
     /**
      * Validate sequence of cron expression.
      *
-     * @param  int\string  $value
+     * @param int|string  $value
```
Comment (peppeocchi) [inline src/GO/Traits/Interval.php]: Can you please re-add the extra space between @param and the type?

File: src/GO/Job.php [hunk 12]
```diff
@@ -2,6 +2,7 @@
 
 use DateTime;
 use Exception;
+use Cron\CronExpression;
```
Comment (peppeocchi) [inline src/GO/Job.php]: This is the line I'd like you to move to the `@var CronExpression` down in the code.

File: src/GO/Job.php (hunk 2 above)
Comment (badyto) [inline src/GO/Job.php]: Okay, I've only changed it because of the return true; I will remove it and change it back to void.

Comment (peppeocchi) [inline src/GO/Job.php]: I am sorry you are right, there is a `return false` just before so your `return true` is absolutely correct, along with the `@return bool`. My apologies can you please add it back?

**Pull request merged**